### PR TITLE
Generalize FamBPMs

### PR DESCRIPTION
**Pull request by fernandohds564**
This PR aims to generalize the class `FamBPMs` to also handle antennas acquisitions. Now, to make acquisitions of an arbitrary set of multiturn signals you must do:

```python3
from siriuspy.devices import FamBPMs

fambpms = FamBPMs()
fambpms.wait_for_connection()
fambpms.mturn_signals2acq = 'XYQAC'
fambpms.mturn_config_acquisition(1000)
fambpms.mturn_update_initial_timestamps()

# trigger the acquisition via timing system

fambpms.mturn_wait_update_timestamps()
data = fambpms.get_mturn_signals()
```
by the end of this snippet, `data` should be a list containing numpy arrays for `'X'`, `'Y'` and `'Q'` positions and `'A'` and `'C'` antennas.

Now it is also possible to select only a subset of the `DEFAULT_PROPERTIES` to be initialized with the object creation when FamBPMs is created only for making triggered acquisitions:
```python3
from siriuspy.devices import FamBPMs

fambpms = FamBPMs()
fambpms_acq = FamBPMs(props2init='acq')

fambpms.pvnames != fambpms_acq.pvnames
```

Besides, we can change the property `mturn_signals2acq` at initialization:
```python3
from siriuspy.devices import FamBPMs

fambpms_xy = FamBPMs()
fambpms_xys = FamBPMs(mturn_signals2acq=list('XYS'))

fambpms_xy.mturn_signals2acq == ['X', 'Y']
fambpms_xys.mturn_signals2acq == ['X', 'Y', 'S']
```

### CODE REVIEW

**Comment by anacso17** [comment]
> fambpms = FamBPMs()
fambpms_acq = FamBPMs(props2init='acq')

great!

File: siriuspy/siriuspy/devices/bpm.py [hunk 1]
```diff
@@ -907,7 +920,9 @@ class FamBPMs(_DeviceSet):
 
     TIMEOUT = 10
     RFFEATT_MAX = 30
+    PROPERTIES_ACQ = BPM.PROPERTIES_ACQ
     PROPERTIES_DEFAULT = BPM.PROPERTIES_DEFAULT
+    ALL_MTURN_SIGNALS2ACQ = ('A', 'B', 'C', 'D', 'X', 'Y', 'Q', 'S')
```
Comment (murilobalves) [inline siriuspy/siriuspy/devices/bpm.py:925]: is there a reason to use capital letters?

**Review by murilobalves — APPROVED** [review]

File: siriuspy/siriuspy/devices/bpm.py (hunk 1 above)
Comment (fernandohds564) [reply siriuspy/siriuspy/devices/bpm.py:925]: I think I'm used to think of BPM Antennas as A, B, C and D. Not a, b, c and d. Then I just kept the same standard for the positions.

File: siriuspy/siriuspy/devices/bpm.py [hunk 2]
```diff
@@ -664,142 +680,139 @@ def acq_channel_str(self):
...
 
     @acq_repeat.setter
     def acq_repeat(self, val):
         """."""
-        self[self.get_propname('ACQTriggerRep-Sel')] = val
+        self['ACQTriggerRep-Sel'] = val
 
     @property
     def acq_update_time(self):
         """."""
```
Comment (xresende) [inline siriuspy/siriuspy/devices/bpm.py:702]: unit?

File: siriuspy/siriuspy/devices/bpm.py [hunk 3]
```diff
@@ -664,142 +680,139 @@ def acq_channel_str(self):
...
-        return self[self.get_propname('ACQUpdateTime-RB')]  / 1e3
+        return self['ACQUpdateTime-RB'] / 1e3
 
     @acq_update_time.setter
     def acq_update_time(self, val):
         """."""
```
Comment (xresende) [inline siriuspy/siriuspy/devices/bpm.py:707]: unit

File: siriuspy/siriuspy/devices/bpm.py [hunk 4]
```diff
@@ -998,77 +1022,46 @@ def get_slow_orbit(self):
...
+                (N, 160), containing the values for the signals acquired.
 
         """
-        orbx, orby = [], []
-        if return_sum:
-            possum = []
+        sigs = [[] for _ in self._mturn_signals2acq]
 
-        mini = None
+        mini = 1000000000000000  # a very large integer
```
Comment (xresende) [inline siriuspy/siriuspy/devices/bpm.py:1035]: i like using language defined constants, instead of arbibrary number. why not `sys.maxsize`?

**Review by xresende — APPROVED** [review]

File: siriuspy/siriuspy/devices/bpm.py (hunk 2 above)
Comment (fernandohds564) [reply siriuspy/siriuspy/devices/bpm.py:702]: done!

File: siriuspy/siriuspy/devices/bpm.py (hunk 3 above)
Comment (fernandohds564) [reply siriuspy/siriuspy/devices/bpm.py:707]: done!

File: siriuspy/siriuspy/devices/bpm.py (hunk 4 above)
Comment (fernandohds564) [reply siriuspy/siriuspy/devices/bpm.py]: done!

**Pull request merged**